File: completable-reactor-runtime/src/main/java/ru/fix/completable/reactor/runtime/gl/GlCoordinates.java
package ru.fix.completable.reactor.runtime.gl;

public interface GlCoordinates {
    /**
     * Payload start point coordinates
     */
    GlCoordinates payload(int x, int y);

    GlCoordinates handler(Vertex vertex, int x, int y);

    GlCoordinates merger(Vertex vertex, int x, int y);

    GlCoordinates router(Vertex vertex, int x, int y);

    GlCoordinates complete(Vertex vertex, int x, int y);
}

File: completable-reactor-runtime/src/main/kotlin/ru/fix/completable/reactor/runtime/gl/GlCoordinates.kt
package ru.fix.completable.reactor.runtime.gl

interface GlCoordinates {
    /**
     * Payload start point coordinates
     */
    fun payload(x: Int, y: Int): GlCoordinates

    fun handler(vertex: Vertex, x: Int, y: Int): GlCoordinates

    fun merger(vertex: Vertex, x: Int, y: Int): GlCoordinates

    fun router(vertex: Vertex, x: Int, y: Int): GlCoordinates

    fun subgraph(vertex: Vertex, x: Int, y: Int): GlCoordinates

    fun complete(vertex: Vertex, x: Int, y: Int): GlCoordinates
}